assets/js/campaign-editors/markdown/menu.js

```diff
@@ -0,0 +1,272 @@
+import { toggleMark } from "prosemirror-commands"
+import { liftListItem, sinkListItem, wrapInList } from "prosemirror-schema-list"
+import { NodeSelection, Plugin, Selection, TextSelection } from "prosemirror-state"
+import { canInsert, hasActiveMark, hasActiveNodeType, toggleBlockType } from "./helpers"
+import { markdownParser } from "./markdown-parser"
+import { schema } from "./markdown-schema"
+import { MenuButton } from "./menu-button"
+
+/** Class for handling editor menus. */
+class MenuView {
+  /**
+   * Initializes a new editor menu.
+   * @param {MenuButton[]} items - MenuButtons
+   * @param {Object} editorView - ProseMirror EditorView
+   */
+  constructor(items, editorView) {
+    this.items = items
+    this.editorView = editorView
+    this.dom = editorView.dom.parentNode.parentNode.querySelector(".wysiwyg--menu")
+
+    items.forEach((item) => item.attach(editorView.dom.parentNode.parentNode))
+
+    this.update()
+
+    this.dom.addEventListener("click", e => {
+      e.preventDefault()
+    })
+
+    this.dom.addEventListener("mousedown", e => {
+      e.stopPropagation()
+      e.preventDefault()
+      editorView.focus()
+      const activatedItem = items.find(({ dom }) => dom.contains(e.target))
+      if (activatedItem) activatedItem.exec(editorView)
+    }, true)
+  }
+
+  update() {
+    this.items.forEach(item => item.update(this.editorView.state))
+  }
+
+  destroy() {
+    this.dom.remove()
```

Comment:
It could be good to remove the "click" and "mousedown" eventListener when the MenuView is destroyed.

Reply:
The event listeners are attached to `this.dom`, so I think they should be removed automatically when the node is remomved, no?

Reply:
Yes, they are, but there are some cases in which we might run into memory leaks.

Especially, if the element is referenced in another place or an anonymous function or closure. So, just for safety, it's best to remove the event listeners before removing the element from the DOM.

Reply:
I tried moving the functions out but they are scoped to the constructor scope, so it wasn’t possible to do it just like that. This PR also doesn’t actually change this code, it’s just moved to a different file :-)